session/kvdb_store.go

```diff
@@ -363,10 +364,39 @@ func (db *BoltStore) GetSession(key *btcec.PublicKey) (*Session, error) {
 	return session, nil
 }
 
-// ListSessions returns all sessions currently known to the store.
+// ListSessions returns all sessions currently known to the store that are in
+// the given states. If no states are provided, all sessions are returned.
 //
 // NOTE: this is part of the Store interface.
-func (db *BoltStore) ListSessions(filterFn func(s *Session) bool) ([]*Session, error) {
+func (db *BoltStore) ListSessions(states ...State) ([]*Session, error) {
```

Comment:
nit:

Given that we have a separate `ListSessionsByType` function, maybe it'd make sense to have a 
`ListSessionsByState` function separately that takes the `states ...State` arg, and then let the `ListSessions` itself take no args an have no filtering?
IMO it just feels a bit weird that the plain `ListSessions` func does take filtering `State(s)` args, while we have another separate function `ListSessionsByType` function.

Reply:
sure yeah can do 👍

Reply:
updated as per suggestion